src/coins.cpp

```diff
@@ -51,7 +51,7 @@ CCoinsMap::iterator CCoinsViewCache::FetchCoin(const COutPoint &outpoint) const
     if (ret->second.coin.IsSpent()) {
         // The parent only has an empty entry for this outpoint; we can consider our
         // version as fresh.
-        ret->second.flags = CCoinsCacheEntry::FRESH;
+        ret->second.AddFlags(CCoinsCacheEntry::FRESH, *ret, m_flagged_head);
```

Comment:
> Should `FRESH`-but-not-`DIRTY` be flagged? I believe those correspond to spent but unmodified coins fetched from the parent cache.

Yes, but only because in this case they are also spent. We erase all spent coins from the cache after doing a non-erasing batch write. There would be no way to track spent-but-not-`DIRTY` otherwise except by scanning the entire cache.

Since the only case where we have `FRESH`-but-not-`DIRTY` is where the coin is spent, it makes sense to me to just add this to the linked list as well. Before proposing this PR I did call the linked list head `m_dirty_head`, but after not finding an efficient way to remove non-dirty spent entries, I decided to call it `m_flagged_head`.

If this makes sense to you I can add a comment here explaining this.

Reply:
see https://github.com/bitcoin/bitcoin/pull/28280#discussion_r1655477509, I think that FRESH-but-not-DIRTY coins may not exist.